sp/src/game/server/hl2/npc_combine.cpp

```diff
@@ -1697,38 +1758,42 @@ void CNPC_Combine::GatherConditions()
 					if ( gpGlobals->curtime - m_flTimePlayerStare >= npc_combine_give_stare_time.GetFloat() &&
 						!IsCurSchedule(SCHED_NEW_WEAPON) && !IsCurSchedule(SCHED_GET_HEALTHKIT) )
 					{
+						bool bAccepted = false;
+
 						// Is it a weapon?
-						if (pHeld->IsBaseCombatWeapon())
+						if (pHeld->IsBaseCombatWeapon() && Weapon_CanUse( pHeld->MyCombatWeaponPointer() ) && IsGiveableWeapon( pHeld->MyCombatWeaponPointer() ))
```

Comment:
Edge case: If pHeld->IsBaseCombatWeapon() is true and pHeld()->MyCombatWeaponPointer() is NULL, this will possibly cause an NPE

(Does IsBaseCombatWeapon() just check the pointer?)

Reply:
I looked in the code - ``IsBaseCombatWeapon()`` is defined as returning true only in ``CBaseCombatWeapon``, right next to where ``*MyCombatWeaponPointer( void )`` is defined as ``{ return this; }``. No problems here